presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java

```diff
@@ -428,6 +436,36 @@ private boolean validateExpressionForGroupBy(Set<Expression> groupByOfMaterializ
         // It must be 1) be selected in the materialized view and 2) not present in GROUP BY node of the materialized view
         return groupByOfMaterializedView.contains(expression) || !materializedViewInfo.getBaseToViewColumnMap().containsKey(expression);
     }
+    /**
+     * This is special-cased for now as COUNT is the only non-associative
+     * function supported by materialized view rewrites. In the future, we will want to
+     * support more non-associative functions and explore more
+     * extensible options.
+     *
+     * Functions in optimized materialized view queries are by default expanded to
+     * func(column_derived_from_func_in_mv). This only works for associative
+     * functions. Count is non-associative: COUNT(x \ union y) != COUNT(Count(x), COUNT(y)).
+     * Rather, COUNT(x \ union y) == SUM(COUNT(x), COUNT(y). This is what we do here.
+     * */
+    private static FunctionCall rewriteCountAsSum(FunctionCall node, Expression derivedColumnName)
+    {
+        if (!node.getName().equals(COUNT)) {
```

Comment:
You already check it when you call so this is redundant - remove one of the places.